src/cogs/logging_message_edit.py

```diff
@@ -1,48 +1,44 @@
-import discord
 from discord.ext import commands
-from datetime import datetime
 from ._settings import log_channel, admin_roles
+from utilities.cog_helpers._embeds import embed_message_edit
 
 
-class logging_messages(commands.Cog):
+class LoggingMessageEdit(commands.Cog):
+    """
+    Simple listener to on_message_edit
+    """
     def __init__(self, bot):
         self.bot = bot
 
     @commands.Cog.listener()
     async def on_message_edit(self, message_before, message_after):
+        """
+        Just checking if the content before is != to the content after.
+        """
         if message_before.content != message_after.content:
+            # This guy here makes sure we use the displayed name inside the guild.
             if message_before.author.nick is None:
                 username = message_before.author
             else:
                 username = message_before.author.nick
 
             author = message_before.author
 
-            embed = discord.Embed(
-                title="<:orange_circle:1043616962112139264> Message Edit",
-                description=f"Edited by {username}\nIn {message_after.channel.mention}",
-                color=discord.Color.dark_orange(),
-                timestamp=datetime.utcnow(),
-            )
-            embed.set_thumbnail(url=author.avatar)
-            embed.add_field(
-                name="Original message: ", value=message_before.content, inline=True
-            )
-
-            embed.add_field(
-                name="After editing: ", value=message_after.content, inline=True
-            )
-
-            logs_channel = await self.bot.fetch_channel(
-                log_channel["chat_log"]
-            )  # ADMIN message log
-
             for role in message_before.author.roles:
-                if role.id in admin_roles.values():
+                if role.id not in admin_roles.values():
+                    """
+                    Dont log admin actions.
+                    This just gets really messy when we are cleaning things up
+                    or doing dodgy business in secret places. 
+                    """
+                    embed = embed_message_edit(username, author, message_before, message_after)
+                    logs_channel = await self.bot.fetch_channel(log_channel["chat_log"])
                     await logs_channel.send(embed=embed)
                     return
```

Comment:
is this behaviour intended? even if a user is an admin, if any of the roles they have are not admin values it would still send a message right? meaning every user that has a non-admin role would have this logging occur

if you wanted to NOT show admins, you should instead do a check like
`is_admin = any(role.id in admin_roles.values() for role in message_before.author.roles)`
and then compare against this variable